api/x509cert_test.go

```diff
@@ -111,24 +115,32 @@ func TestGetX509CACertificate(t *testing.T) {
 			KeyMeta:      &proto.KeyMeta{Identifier: "x509id2"},
 			expectedCert: nil,
 		},
+		"requestTimeout": {
+			ctx:          timeoutCtx,
+			KeyUsages:    combineKeyUsage,
+			KeyMeta:      &proto.KeyMeta{Identifier: "x509id1"},
+			expectedCert: nil,
+		},
```

Comment:
can we add a test for context cancelled by client, while the request is being processed?